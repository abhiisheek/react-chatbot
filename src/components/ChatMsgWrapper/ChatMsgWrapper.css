@import '../_colors.css';
@import '../_variables.css';

.chatMsgWrapper {
  position: relative;
  border-radius: 6px;
  border: 1px solid var(--border-color);
  padding: 10px 10px;
  margin: 0 5px;
  max-width: 75%;
  box-shadow: var(--shadow-2p);
  word-wrap: break-word;
}

.chatMsgWrapper a {
  color: var(--link-text-color);
  font-size: var(--font-size-tiny);
}

.chatMsgWrapper a:hover {
  color: var(--link-text-hover-color);
  text-decoration: none;
}
