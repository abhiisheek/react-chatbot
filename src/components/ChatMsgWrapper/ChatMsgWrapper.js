import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import withWrapper from '../hocs/WithWrapper';
import styles from './ChatMsgWrapper.css';

const ChatMsgWrapperComp = withWrapper(styles.chatMsgWrapper, 'ChatMsg');

/** Component for wrapper all other type component into nice chat message(capsule) look */
class ChatMsgWrapper extends PureComponent {
  render() {
    const { children, className } = this.props;
    return <ChatMsgWrapperComp children={children} className={className} />;
  }
}

ChatMsgWrapper.displayName = ChatMsgWrapperComp.displayName;

ChatMsgWrapper.propTypes = {
  /** Children component that needs to be wrapped */
  children: PropTypes.element.isRequired,
  /** CSS class for customisation */
  className: PropTypes.string
};

export default ChatMsgWrapper;
