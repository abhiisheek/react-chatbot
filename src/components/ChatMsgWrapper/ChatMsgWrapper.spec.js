import React from 'react';
import renderer from 'react-test-renderer';

import ChatMsgWrapper from './ChatMsgWrapper';

describe('ChatMsgWrapper', () => {
  it('Should have displayName set on the component', () => {
    expect(ChatMsgWrapper.displayName).toBeDefined();
  });

  it('Snapshot test', () => {
    const wrapper = renderer.create(
      <ChatMsgWrapper>
        <div>{'hi'}</div>
      </ChatMsgWrapper>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
