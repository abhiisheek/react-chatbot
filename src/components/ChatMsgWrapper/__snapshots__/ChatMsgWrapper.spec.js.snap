// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMsgWrapper Snapshot test 1`] = `
<div
  className="chatMsgWrapper"
>
  <div>
    hi
  </div>
</div>
`;
